examples/sglang_bench/bench_sharegpt.py

```
@@ -3,41 +3,224 @@
+    ("A100", "a100"),
+    ("MI355", "mi355x"),
+    ("MI325", "mi325x"),
+    ("MI300", "mi300x"),
+]
+
+
+def detect_gpu_type() -> str:
+    """Auto-detect GPU type from nvidia-smi or rocm-smi, returning a sgl-cookbook hardware ID."""
+    try:
+        result = subprocess.run(
+            ["nvidia-smi", "--query-gpu=name", "--format=csv,noheader,nounits"],
```

Ruff (0.15.9)

[error] Starting a process with a partial executable path

(S607)